### PR TITLE
schemabuilder: Parallelize filter and sort functions

### PR DESCRIPTION
**Pull request by berfarah**
Because we might be resolving many, many nodes, applySort and applyFilter need to be parallelized.

### CODE REVIEW

File: graphql/schemabuilder/pagination.go [hunk 1]
```diff
@@ -397,31 +398,47 @@ func (c *connectionContext) applyTextFilter(ctx context.Context, nodes []interfa
...
-				return nil, err
-			}
+	nodesToKeep := make([]bool, len(nodes))
 
-			// Only strings are allowed for FilterText fields.
-			textString, ok := text.(string)
-			if !ok {
-				return nil, fmt.Errorf("filter %s returned %T, must be a string", name, text)
-			}
+	g, _ := errgroup.WithContext(ctx)
```
Comment (stephen) [inline graphql/schemabuilder/pagination.go]: why don't we propagate the `ctx` returned?

Comment (berfarah) [reply graphql/schemabuilder/pagination.go]: Done

File: graphql/schemabuilder/pagination.go [hunk 2]
```diff
@@ -397,31 +398,47 @@ func (c *connectionContext) applyTextFilter(ctx context.Context, nodes []interfa
...
+	nodesToKeep := make([]bool, len(nodes))
 
-			// Only strings are allowed for FilterText fields.
-			textString, ok := text.(string)
-			if !ok {
-				return nil, fmt.Errorf("filter %s returned %T, must be a string", name, text)
-			}
+	g, ctx := errgroup.WithContext(ctx)
+	for unscopedI, node := range nodes {
+		i, scopedNode := unscopedI, node
```
Comment (willhug) [inline graphql/schemabuilder/pagination.go]: can we be all "scoped" or all "unscoped" instead of one "scoped" and one "unscoped" lol

Comment (berfarah) [reply graphql/schemabuilder/pagination.go]: Yeah, done

**Pull request merged**